manual/manual/manual.tex

```diff
@@ -3,17 +3,111 @@
 \usepackage[utf8]{inputenc}
 \usepackage[T1]{fontenc}
 % HEVEA\@def@charset{UTF-8}%
-\usepackage{alltt}
 \usepackage{fullpage}
 \usepackage{syntaxdef}
 \usepackage{multind}
 \usepackage{html}
 \usepackage{textcomp}
-\usepackage{caml-sl}
 \usepackage{ocamldoc}
 \usepackage{xspace}
+\usepackage{color}
+
+% Package for code examples:
+\usepackage{listings}
+\usepackage{alltt}
+\usepackage{lmodern}% for supporting bold ttfamily in code examples
+\usepackage[normalem]{ulem}% for underlining errors in code examples
+
 
 \input{macros.tex}
+\newcommand{\hash}{\#}
+\lstnewenvironment{camloutput}{
+  \lstset{
+    basicstyle=\small\ttfamily\slshape,
+    showstringspaces=false,
+    language=caml,
+    escapeinside={$}{$},
```

Comment:
I think that `$` is not a very good choice of escape characters, because they are what the `mathescape` option uses to escape directly into math mode (we don't use that option currently but still the choice is confusing). Maybe you could use `\{...\}` for example, or `#(...)#`?

Reply:
I had some troubles with n-graphs escape (and `$` only appear once in the current code example). Let me recheck that.

Reply:
Examples I find on the web tend to avoid `\` and `{`. I've seen `(*..*)` proposed which is obviously a terrible choice, but also for example `@*...*@` or `%*...*%` which look reasonable enough.

Reply:
After few tries, there is a conflict somewhere and using n-grams does not seem to work with the html backend. I am thus keeping `$` for now.